flink-table/flink-table-planner/src/main/java/org/apache/flink/table/expressions/rules/ResolveCallByArgumentsRule.java

```diff
@@ -86,21 +86,31 @@ private Expression performTypeResolution(CallExpression call, List<TypeInformati
 
 		private Expression validateArguments(CallExpression call, PlannerExpression plannerCall) {
 			if (!plannerCall.valid()) {
-				final String errorMessage;
-				ValidationResult validationResult = plannerCall.validateInput();
-				if (validationResult instanceof ValidationFailure) {
-					errorMessage = ((ValidationFailure) validationResult).message();
-				} else {
-					errorMessage = String.format("Invalid arguments %s for function: %s",
-						call.getChildren(),
-						call.getFunctionDefinition().getName());
-				}
-				throw new ValidationException(errorMessage);
+				throw new ValidationException(getValidationErrorMessage(plannerCall));
 			}
-
 			return call;
 		}
 
+		/**
+		 * Return the validation error message of this {@link PlannerExpression} or return the
+		 * validation error message of it's children if it passes the validation. Return an empty
+		 * string if all validation succeeded.
+		 */
+		private String getValidationErrorMessage(PlannerExpression plannerCall) {
```

Comment:
One small comment, let's use the `Optional` to mark that method might not return results.